sdks/python/apache_beam/runners/common.py

```diff
@@ -968,6 +973,9 @@ def process_with_sized_restriction(self, windowed_value):
     (element, (restriction, estimator_state)), _ = windowed_value.value
     watermark_estimator = (
         self.do_fn_invoker.invoke_create_watermark_estimator(estimator_state))
+    if restriction is None:
```

Comment:
Same issue as in java, we should allow None restrictions and instead produce 0 outputs.